tasks/main.yml

```
@@ -1277,11 +1324,11 @@
 
         - name: Verify that primary instance hosts is available
           assert:
-            that: ansible_play_hosts |
-                map('extract', hostvars, '__mssql_ha_is_primary') |
-                select('match', '^[Tt]rue$') |
-                list |
-                length == 1
+            that: ansible_play_hosts
+              | map('extract', hostvars, '__mssql_ha_is_primary')
+              | select('equalto', true)
```

```suggestion
              | select
```
`select` will return the value if it evaluates to `true` in a boolean context